provider/identity_provider_resource.go

```diff
@@ -283,8 +315,29 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 			htpasswdBuilder.Password(state.HTPasswd.Password.Value)
 		}
 		builder.Htpasswd(htpasswdBuilder)
+    case state.Gitlab != nil:
+		builder.Type(cmv1.IdentityProviderTypeGitlab)
+		gitlabBuilder := cmv1.NewGitlabIdentityProvider()
+		if !state.Gitlab.CA.Unknown && !state.Gitlab.CA.Null {
+			gitlabBuilder.CA(state.Gitlab.CA.Value)
+		}
+		gitlabBuilder.ClientID(state.Gitlab.ClientID.Value)
+		gitlabBuilder.ClientSecret(state.Gitlab.ClientSecret.Value)
+		u, err := url.ParseRequestURI(state.Gitlab.URL.Value)
+		if err != nil && u.Scheme != "" && u.RawQuery != "" && u.Fragment != "" {
+			response.Diagnostics.AddError(
+				"Expected a valid GitLab provider URL",
+				fmt.Sprintf(
+					"Can't build identity provider with name '%s': %v",
```

Comment:
"Expected a valid GitLab provider URL: to use an https:// scheme, must not have query parameters and not have a fragment....."